File: AI.Core/OperatorsImplimentation/BasicOperators/Minus.cs
using AI.Core.ChannelsImplimentation;

namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    public class Minus : Operator
    {
        public Minus(Entity thisEntity)
            : base(thisEntity)
        {
            EnterContactsCount = 2;
        }

        public override void Action()
        {
            var value = EnterContacts[0].Value - EnterContacts[1].Value;
            ExitContacts.SetValue(value);
        }
    }
}

File: NativeCore/OperatorsImplimentation/BasicOperators/Minus.cpp
#include "Minus.h"

using namespace AI::Core::ChannelsImplimentation;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				Minus::Minus(Entity *thisEntity) : Operator(thisEntity)
				{
					setEnterContactsCount(2);
				}

				void Minus::Action()
				{
					auto value = EnterContacts[0]->Value - EnterContacts[1]->Value;
					ExitContacts->SetValue(value);
				}
			}
		}
	}
}
